experimental/beacon_sim/ethans_super_cool_planner.hh

```diff
@@ -0,0 +1,66 @@
+
+
+#include "experimental/beacon_sim/make_belief_updater.hh"
+#include "experimental/beacon_sim/robot_belief.hh"
+#include "planning/breadth_first_search.hh"
+#include "planning/probabilistic_road_map.hh"
+
+#include <iostream>
+
+/*
+BFS: given graph, search for paths
+BeliefRoadMap:  given graph, beacons, do belief updates
+BeliefRoadMapPlanner: using these belief updates and search, find a nice path
+
+*/
+
+namespace robot::experimental::beacon_sim {
+
+struct Candidate {
+    RobotBelief belief;             // where the robot thinks it is
+    std::vector<int> path_history;  // the path that got it here
+};
+
+
+struct RollOutArgs {
+    unsigned int num_roll_outs;
+};
+
+
+std::vector<Candidate> rollout ( const planning::RoadMap& map, 
+                                 const std::function<bool(const Candidate&, int)>& terminate_rollout,
+                                 const Candidate& candidate, 
+                                 const std::function<std::vector<planning::Successor<int>>(const int)>& successor_function,
+                                 const planning::BeliefUpdater<RobotBelief>& belief_updater,
+                                 const RollOutArgs& roll_out_args );
```

Comment:
I'm a little surprised by this interface. I could be wrong, but I thought that we wouldn't need to explicitly reason about the belief during the rollout. I'm willing to be convinced though.